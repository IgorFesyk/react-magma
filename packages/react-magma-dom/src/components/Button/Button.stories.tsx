import React from 'react';
import { Button } from '.';

export default {
  component: Button,
  title: 'Button',
};

export const Default = () => {
  return <Button>Default</Button>;
};
