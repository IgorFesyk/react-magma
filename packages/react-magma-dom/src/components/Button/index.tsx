import * as React from 'react';
import { ButtonCore } from 'react-magma-core';
import {
  StyledButton,
  ButtonProps,
  ButtonColor,
  ButtonShape,
  ButtonSize,
  ButtonTextTransform,
  ButtonVariant,
  ButtonIconPostition
} from '../StyledButton';
import styled from '@emotion/styled';
import { IconProps } from '../Icon/utils';
import { Omit } from '../utils';

export interface IconOnlyButtonProps extends Omit<ButtonProps, 'children'> {
  icon: React.ReactElement<IconProps>;
  ariaLabel: string;
}

interface IconTextButtonProps extends ButtonProps {
  icon: React.ReactElement<IconProps>;
  children: React.ReactChild | React.ReactChild[];
  iconPosition?: ButtonIconPostition;
}

export type MergedButtonProps =
  | ButtonProps
  | IconTextButtonProps
  | IconOnlyButtonProps;

export interface SpanProps {
  size?: ButtonSize;
}

const SpanTextLeft = styled.span<SpanProps>`
  padding-right: ${props => (props.size === 'large' ? '15px' : '10px')};
`;

const SpanTextRight = styled.span<SpanProps>`
  padding-left: ${props => (props.size === 'large' ? '15px' : '10px')};
`;

function getIconSize(size) {
  switch (size) {
    case 'large':
      return 24;
    case 'small':
      return 14;
    default:
      return 18;
  }
}

function getIconWithTextSize(size) {
  switch (size) {
    case 'large':
      return 20;
    case 'small':
      return 12;
    default:
      return 16;
  }
}

function instanceOfIconText(object: any): object is IconTextButtonProps {
  return 'icon' in object && 'children' in object;
}

function instanceOfIconOnly(object: any): object is IconOnlyButtonProps {
  return 'icon' in object && !('children' in object);
}

export const Button: React.FunctionComponent<
  MergedButtonProps
> = React.forwardRef((props: MergedButtonProps, ref: any) => (
  <ButtonCore onClick={props.onClick}>
    {({ onClick }) => {
      let icon;
      let iconPosition;
      let children;
      const {
        ariaExpanded,
        ariaLabel,
        as,
        autoFocus,
        disabled,
        inverse,
        block,
        className,
        color,
        href,
        id,
        testId,
        shape,
        size,
        style,
        variant,
        textTransform,
        to
      } = props;

      if (instanceOfIconOnly(props)) {
        icon = props.icon;
      } else if (instanceOfIconText(props)) {
        icon = props.icon;
        iconPosition = props.iconPosition;
        children = props.children;
      } else {
        children = props.children;
      }

      if (icon && children) {
        return (
          <StyledButton
            id={id}
            testId={testId}
            ref={ref}
            ariaLabel={ariaLabel}
            ariaExpanded={ariaExpanded}
            as={as}
            autoFocus={autoFocus}
            onClick={onClick}
            block={block}
            color={color ? color : ButtonColor.primary}
            className={className}
            disabled={disabled}
            inverse={inverse}
            href={href}
            shape={shape ? shape : ButtonShape.fill}
            size={size ? size : ButtonSize.medium}
            style={style}
            textTransform={
              textTransform ? textTransform : ButtonTextTransform.uppercase
            }
            to={to}
            variant={variant ? variant : ButtonVariant.solid}
          >
            {iconPosition === ButtonIconPostition.right && (
              <SpanTextLeft size={size}>{children} </SpanTextLeft>
            )}
            {React.Children.only(
              React.cloneElement(icon, { size: getIconWithTextSize(size) })
            )}
            {iconPosition !== ButtonIconPostition.right && (
              <SpanTextRight size={size}>{children}</SpanTextRight>
            )}
          </StyledButton>
        );
      } else if (icon && !children) {
        return (
          <StyledButton
            id={id}
            testId={testId}
            ref={ref}
            ariaExpanded={ariaExpanded}
            ariaLabel={ariaLabel}
            as={as}
            autoFocus={autoFocus}
            onClick={onClick}
            className={className}
            color={color ? color : ButtonColor.primary}
            disabled={disabled}
            iconOnly
            inverse={inverse}
            href={href}
            shape={shape ? shape : ButtonShape.round}
            size={size ? size : ButtonSize.medium}
            style={style}
            to={to}
            variant={variant ? variant : ButtonVariant.solid}
          >
            {React.Children.only(
              React.cloneElement(icon, { size: getIconSize(size) })
            )}
          </StyledButton>
        );
      }

      return (
        <StyledButton
          id={id}
          testId={testId}
          ref={ref}
          as={as}
          ariaExpanded={ariaExpanded}
          ariaLabel={ariaLabel}
          autoFocus={autoFocus}
          onClick={onClick}
          block={block}
          color={color ? color : ButtonColor.primary}
          className={className}
          disabled={disabled}
          href={href}
          inverse={inverse}
          shape={shape ? shape : ButtonShape.fill}
          size={size ? size : ButtonSize.medium}
          style={style}
          textTransform={
            textTransform ? textTransform : ButtonTextTransform.uppercase
          }
          to={to}
          variant={variant ? variant : ButtonVariant.solid}
        >
          {children}
        </StyledButton>
      );
    }}
  </ButtonCore>
));
