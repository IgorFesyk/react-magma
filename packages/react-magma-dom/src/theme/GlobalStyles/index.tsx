import * as React from 'react';
import { Global, css } from '@emotion/core';
import { ThemeContext } from '../ThemeContext';

function getStyles(theme) {
  return css`
    @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,300i,400,400i,600,600i,700,700i,800,800i');

    *,
    *:before,
    *:after {
      box-sizing: border-box;
    }

    *:focus {
      outline: 2px dotted ${theme.colors.pop02};
      outline-offset: 3px;
      transition: outline 0.1s linear;
    }

    html {
      scroll-behavior: smooth;
    }

    html,
    body {
      margin: 0;
      padding: 0;
    }

    body {
      background: ${theme.colors.neutral08};
      color: ${theme.colors.neutral02};
      font-family: ${theme.bodyFont};
      font-style: normal;
      font-weight: 400;
      font-size: 15px;
      line-height: 22px;
    }

    a {
      color: ${theme.colors.primary};
      cursor: pointer;
      text-decoration: underline;

      &:hover,
      &:focus {
        color: ${theme.colors.foundation01};
        text-decoration: none;
      }
    }
  `;
}

export const GlobalStyles: React.FunctionComponent = () => {
  return (
    <ThemeContext.Consumer>
      {theme => <Global styles={getStyles(theme)} />}
    </ThemeContext.Consumer>
  );
};
