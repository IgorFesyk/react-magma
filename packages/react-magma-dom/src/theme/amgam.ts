interface Colors {
  primary: string;
  foundation01: string;
  foundation02: string;
  foundation03: string;
  foundation04: string;
  pop01: string;
  pop02: string;
  pop03: string;
  pop04: string;
  success01: string;
  success02: string;
  danger: string;
  neutral01: string;
  neutral02: string;
  neutral03: string;
  neutral04: string;
  neutral05: string;
  neutral06: string;
  neutral07: string;
  neutral08: string;
  disabledText: string;
  disabledInverseText: string;
}

export interface ThemeInterface {
  bodyFont: string;
  headingFont: string;
  colors: Colors;
  sizeXs: string;
  sizeSm: string;
  sizeMd: string;
  sizeLg: string;
}

export const amgam = {
  bodyFont: '"Open Sans", sans-serif',
  headingFont: 'gnuolane, sans-serif',

  colors: {
    primary: '#FF9D67', // link color blue
    foundation01: '#FFC79A',
    foundation02: '#FF9D67',
    foundation03: '#ff9d67',
    foundation04: '#006298',
    pop01: '#1FFFB2',
    pop02: '#0D56FF',
    pop03: '#F2A900',
    pop04: '#0038D3',
    success01: '#C57DFF',
    success02: '#AC44FF',
    danger: '#FFD9C1',
    neutral01: '#00263E',
    neutral02: '#C0C0C0', // main dark grey text color
    neutral03: '#A8A8A8',
    neutral04: '#8D8D8D',
    neutral05: '#404040',
    neutral06: '#202020',
    neutral07: '#080808',
    neutral08: '#000000', // white
    disabledText: 'rgba(114, 114, 114, 0.6)',
    disabledInverseText: 'rgba(255, 255, 255, 0.25)'
  },

  // screen sizes
  sizeXs: '576px',
  sizeSm: '768px',
  sizeMd: '992px',
  sizeLg: '1200px'
};
