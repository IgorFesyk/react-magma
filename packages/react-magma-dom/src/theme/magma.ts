export interface Colors {
  primary: string;
  foundation01: string;
  foundation02: string;
  foundation03: string;
  foundation04: string;
  pop01: string;
  pop02: string;
  pop03: string;
  pop04: string;
  success01: string;
  success02: string;
  danger: string;
  neutral01: string;
  neutral02: string;
  neutral03: string;
  neutral04: string;
  neutral05: string;
  neutral06: string;
  neutral07: string;
  neutral08: string;
  a11y01: string;
  disabledText: string;
  disabledInverseText: string;
  toggleBoxShadow: string;
}

export interface ThemeInterface {
  bodyFont: string;
  headingFont: string;
  colors: Colors;
  sizeXs: string;
  sizeSm: string;
  sizeMd: string;
  sizeLg: string;
}

export const magma = {
  bodyFont: '"Open Sans", sans-serif',
  headingFont: '"Open Sans", sans-serif',

  colors: {
    primary: '#006298', // link color blue
    foundation01: '#003865',
    foundation02: '#006298',
    foundation03: '#00A9E0',
    foundation04: '#71C5E8',
    pop01: '#E0004D',
    pop02: '#FA6600',
    pop03: '#F2A900',
    pop04: '#FFC72C',
    success01: '#3A8200',
    success02: '#48A200',
    danger: '#E70000',
    neutral01: '#00263E',
    neutral02: '#3F3F3F', // main dark grey text color
    neutral03: '#575757',
    neutral04: '#727272',
    neutral05: '#BFBFBF',
    neutral06: '#DFDFDF',
    neutral07: '#F7F7F7',
    neutral08: '#FFFFFF', // white
    a11y01: '#8f8f8f', // lightest gray that meets 3:1 contrast ratio
    disabledText: 'rgba(114, 114, 114, 0.6)',
    disabledInverseText: 'rgba(255, 255, 255, 0.25)',
    toggleBoxShadow:
      '0 2px 2px -1px rgba(0, 0, 0, 0.3), 0 0 4px 0 rgba(0, 0, 0, 0.24),0 0 5px 0 rgba(0, 0, 0, 0.22)'
  },

  // screen sizes
  sizeXs: '600px',
  sizeSm: '768px',
  sizeMd: '1024px',
  sizeLg: '1200px'
};
