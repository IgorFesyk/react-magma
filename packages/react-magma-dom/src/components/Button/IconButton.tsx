import * as React from 'react';
import { ButtonCore } from 'react-magma-core';
import { ButtonProps, StyledButton } from './Button';
import { Icon } from '../Icon/Icon';
import styled from 'styled-components';

enum ButtonTextPostition {
  left = 'left',
  right = 'right'
}

export interface IconButtonProps extends ButtonProps {
  icon: string;
  label?: string;
  textPosition?: ButtonTextPostition;
}

const StyledIconButton = styled(StyledButton)<IconButtonProps>`
  border-radius: ${props => {
    switch (props.shape) {
      case 'fill':
        return '5px';
      case 'leftCap':
        return '5px 0 0 5px';
      case 'rightCap':
        return '0 5px 5px 0';
      default:
        return '100%';
    }
  }};
  display: inline-flex;
  justify-content: center;
  line-height: 1;
  min-width: 0;
  padding: 0;

  height: ${props => {
    switch (props.size) {
      case 'large':
        return '44px';
      case 'small':
        return '28px';
      default:
        return '37px';
    }
  }};

  width: ${props => {
    switch (props.size) {
      case 'large':
        return '44px';
      case 'small':
        return '28px';
      default:
        return '37px';
    }
  }};
`;

const SpanTextLeft = styled.span`
  padding-right: 10px;
`;

const SpanTextRight = styled.span`
  padding-left: 10px;
`;

function getIconSize(size) {
  switch (size) {
    case 'large':
      return 24;
    case 'small':
      return 14;
    default:
      return 18;
  }
}

export const IconButton: React.FunctionComponent<IconButtonProps> = (
  props: IconButtonProps
): JSX.Element => (
  <ButtonCore handleClick={props.handleClick}>
    {({ handleClick }) => {
      const {
        autoFocus,
        label,
        disabled,
        icon,
        inverse,
        color,
        shape,
        size,
        type,
        textTransform,
        textPosition
      } = props;

      if (textPosition) {
        return (
          <StyledButton
            autoFocus={autoFocus}
            onClick={handleClick}
            color={color ? color : 'primary'}
            disabled={disabled}
            inverse={inverse}
            shape={shape ? shape : 'fill'}
            size={size ? size : 'medium'}
            textTransform={textTransform ? textTransform : 'uppercase'}
            type={type ? type : 'solid'}
          >
            {textPosition === ButtonTextPostition.left && (
              <SpanTextLeft>{label} </SpanTextLeft>
            )}
            <Icon size={16} type={icon} />
            {textPosition === ButtonTextPostition.right && (
              <SpanTextRight>{label}</SpanTextRight>
            )}
          </StyledButton>
        );
      }

      return (
        <StyledIconButton
          aria-label={label}
          autoFocus={autoFocus}
          onClick={handleClick}
          color={color ? color : 'primary'}
          disabled={disabled}
          inverse={inverse}
          shape={shape ? shape : 'round'}
          size={size ? size : 'medium'}
          type={type ? type : 'solid'}
        >
          <Icon size={getIconSize(size ? size : 'medium')} type={icon} />
        </StyledIconButton>
      );
    }}
  </ButtonCore>
);

export default IconButton;
