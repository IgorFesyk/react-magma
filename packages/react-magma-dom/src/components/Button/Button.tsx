import * as React from 'react';
import { ButtonCore } from 'react-magma-core';
const styled = require('styled-components').default;
import { magma } from '../../theme/magma';

enum ButtonType {
  solid, //default
  outline,
  link
}

enum ButtonColor {
  primary, //default
  secondary,
  success,
  danger
}

enum ButtonShape {
  fill, //default
  leftCap,
  rightCap,
  round
}

enum ButtonSize {
  large,
  medium, //default
  small
}

export interface ButtonProps {
  allCaps?: boolean;
  autoFocus?: boolean;
  text: string;
  handleClick: () => void;
  color?: ButtonColor;
  disabled?: boolean;
  inverse?: boolean;
  shape?: ButtonShape;
  size?: ButtonSize;
  type?: ButtonType;
}

const StyledButton = styled.button`
  display: inline-block;
  font-family: ${magma.bodyFont};
  line-height: 1.46666667;
  margin: 5px;
  overflow: hidden;
  position: relative;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  white-space: nowrap;

  border-radius: ${props => {
    // Button shape
    switch (props.shape) {
      case 'leftCap':
        return '5px 0 0 5px';
      case 'rightCap':
        return '0 5px 5px 0';
      case 'round':
        return '100%';
      default:
        return '5px';
    }
  }};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  text-transform: ${props => (props.allCaps === false ? 'none' : 'uppercase')};

  font-size: ${props => {
    // Button size
    switch (props.size) {
      case 'large':
        return '1.125rem';
      case 'small':
        return '.750rem';
      default:
        return '.875rem';
    }
  }};
  font-weight: ${props => (props.size === 'large' ? 500 : 600)};
  height: ${props => {
    // Button size
    switch (props.size) {
      case 'large':
        return '45px';
      case 'small':
        return '29px';
      default:
        return '37px';
    }
  }};
  padding: ${props => {
    // Button size
    switch (props.size) {
      case 'large':
        return '0 20px';
      case 'small':
        return '0 10px';
      default:
        return '0 15px';
    }
  }};

  background: ${props => {
    // Button color, type, inverse
    if (props.type === 'link' || props.type === 'outline') {
      return 'none';
    }
    if (props.disabled) {
      return magma.colors.neutral06;
    }
    if (props.inverse) {
      return magma.colors.neutral08;
    }
    switch (props.color) {
      case 'secondary':
        return magma.colors.neutral08;
      case 'success':
        return magma.colors.success01;
      case 'danger':
        return magma.colors.danger;
      default:
        return magma.colors.primary;
    }
  }};
  border: ${props => (props.type === 'link' ? '0' : '2px solid')};
  border-color: ${props => {
    // Button color, type, inverse
    if (props.disabled) {
      return magma.colors.neutral06;
    }
    if (props.inverse) {
      return magma.colors.neutral08;
    }
    if (props.color === 'secondary') {
      return magma.colors.neutral05;
    }
    if (props.type === 'solid') {
      switch (props.color) {
        case 'success':
          return magma.colors.success01;
        case 'danger':
          return magma.colors.danger;
        default:
          return magma.colors.primary;
      }
    }
  }};
  color: ${props => {
    // Button color, type, inverse
    if (props.disabled) {
      return magma.colors.neutral04;
    }
    if (
      (!props.inverse && props.type === 'solid') ||
      (props.inverse && props.type !== 'solid')
    ) {
      if (props.color === 'secondary' && !props.inverse) {
        return magma.colors.neutral02;
      }
      return magma.colors.neutral08;
    }
    switch (props.color) {
      case 'secondary':
        return magma.colors.neutral02;
      case 'success':
        return magma.colors.success01;
      case 'danger':
        return magma.colors.danger;
      default:
        return magma.colors.primary;
    }
  }};

  &:before {
    background: ${props =>
      props.color === 'secondary'
        ? 'rgba(63,63,63,1)'
        : 'rgba(255, 255, 255, 1)'};
    content: '';
    height: 200%;
    left: -50%;
    opacity: 0;
    position: absolute;
    top: -50%;
    transition: 0.2s;
    width: 200%;
  }

  &:hover {
    &:before {
      opacity: 0.1;
    }
  }
`;

export const Button: React.FunctionComponent<ButtonProps> = (
  props: ButtonProps
): JSX.Element => (
  <ButtonCore handleClick={props.handleClick}>
    {({ handleClick }) => {
      const {
        autoFocus,
        text,
        disabled,
        inverse,
        allCaps,
        color,
        shape,
        size,
        type
      } = props;

      return (
        <StyledButton
          allCaps={allCaps}
          autoFocus={autoFocus}
          onClick={handleClick}
          color={color ? color : 'primary'}
          disabled={disabled}
          inverse={inverse}
          shape={shape ? shape : 'fill'}
          size={size ? size : 'medium'}
          type={type ? type : 'solid'}
        >
          {text}
        </StyledButton>
      );
    }}
  </ButtonCore>
);

export default Button;
