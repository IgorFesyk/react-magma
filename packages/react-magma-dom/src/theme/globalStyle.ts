import { createGlobalStyle } from 'styled-components';
import { magma } from './magma';

// tslint:disable:no-unused-expression
export const Theme = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,300i,400,400i,600,600i,700,700i,800,800i');
  @import url('https://use.typekit.net/rwr6vzk.css');

  *,
  *:before,
  *:after {
    box-sizing: border-box;
  }

  & body {
    color: ${magma.colors.neutral02};
  	font-family: ${magma.bodyFont};
  	font-style: normal;
  	font-weight: 400;
  	font-size: 15px;
  	line-height: 22px;
  }

  a {
    color: ${magma.colors.primary};
    cursor: pointer;
    text-decoration: none;

    &:hover,
    &:focus {
      color: ${magma.colors.foundation01};
      text-decoration: underline;
    }
  }
`;
