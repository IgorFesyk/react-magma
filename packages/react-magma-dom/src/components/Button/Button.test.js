import React from 'react';
import { axe } from 'jest-axe';
import { Button } from './Button';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { magma } from '../../theme/magma';
import {
  ButtonColor,
  ButtonShape,
  ButtonSize,
  ButtonTextTransform,
  ButtonVariant
} from './StyledButton';

const TEXT = 'Test Text';

const BASE_BUTTON_PROPS = {
  handleClick: jest.fn()
};

const renderButton = (myProps = {}) => {
  const props = {
    ...BASE_BUTTON_PROPS,
    ...myProps
  };

  return render(<Button {...props}>{TEXT}</Button>);
};

describe('Button', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render a button with the passed in text', () => {
    const { getByText } = renderButton();

    expect(getByText(TEXT)).not.toBeNull();
  });

  describe('Snapshot', () => {
    it('should render with non-default props', () => {
      const { container } = render(
        <div>
          <Button {...BASE_BUTTON_PROPS} color={ButtonColor.secondary}>
            {TEXT}
          </Button>
          <Button {...BASE_BUTTON_PROPS} shape={ButtonShape.round}>
            {TEXT}
          </Button>
          <Button {...BASE_BUTTON_PROPS} size={ButtonSize.small}>
            {TEXT}
          </Button>
          <Button
            {...BASE_BUTTON_PROPS}
            textTransform={ButtonTextTransform.none}
          >
            {TEXT}
          </Button>
          <Button {...BASE_BUTTON_PROPS} variant={ButtonVariant.outline}>
            {TEXT}
          </Button>
        </div>
      );

      expect(container).toMatchSnapshot();
    });
  });

  it('Does not violate accessibility standards', () => {
    const { container } = renderButton();
    return axe(container.innerHTML).then(result => {
      return expect(result).toHaveNoViolations();
    });
  });
});
