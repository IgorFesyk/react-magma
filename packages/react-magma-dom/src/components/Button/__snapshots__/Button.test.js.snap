// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Base Button Snapshot should render with large size 1`] = `
<div>
  <button
    class="css-mu0rhi e183qit0"
    color="primary"
    shape="fill"
  >
    Test Text
  </button>
</div>
`;

exports[`Button Base Button Snapshot should render with small size 1`] = `
<div>
  <button
    class="css-1b9wjjc e183qit0"
    color="primary"
    shape="fill"
  >
    Test Text
  </button>
</div>
`;

exports[`Button Base Button Snapshot should render with updated color 1`] = `
<div>
  <button
    class="css-xh5nj e183qit0"
    color="secondary"
    shape="fill"
  >
    Test Text
  </button>
</div>
`;

exports[`Button Base Button Snapshot should render with updated shape 1`] = `
<div>
  <button
    class="css-15f1fu6 e183qit0"
    color="primary"
    shape="round"
  >
    Test Text
  </button>
</div>
`;

exports[`Button Base Button Snapshot should render with updated textTransform 1`] = `
<div>
  <button
    class="css-9tyo2z e183qit0"
    color="primary"
    shape="fill"
  >
    Test Text
  </button>
</div>
`;

exports[`Button Base Button Snapshot should render with updated variant 1`] = `
<div>
  <button
    class="css-3a4sd6 e183qit0"
    color="primary"
    shape="fill"
  >
    Test Text
  </button>
</div>
`;

exports[`Button Icon Only Button Snapshot should render with large size 1`] = `
<div>
  <button
    aria-label="Check Icon"
    class="css-ck9427 e183qit0"
    color="primary"
    shape="round"
  >
    <svg
      class="icon"
      fill="currentColor"
      height="14"
      viewBox="0 0 29.37 21.88"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon Only Button Snapshot should render with small size 1`] = `
<div>
  <button
    aria-label="Check Icon"
    class="css-ck9427 e183qit0"
    color="primary"
    shape="round"
  >
    <svg
      class="icon"
      fill="currentColor"
      height="14"
      viewBox="0 0 29.37 21.88"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon Only Button Snapshot should render with updated color 1`] = `
<div>
  <button
    aria-label="Check Icon"
    class="css-1c2yoq5 e183qit0"
    color="secondary"
    shape="round"
  >
    <svg
      class="icon"
      fill="currentColor"
      height="18"
      viewBox="0 0 29.37 21.88"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon Only Button Snapshot should render with updated shape 1`] = `
<div>
  <button
    aria-label="Check Icon"
    class="css-1kwullk e183qit0"
    color="primary"
    shape="fill"
  >
    <svg
      class="icon"
      fill="currentColor"
      height="18"
      viewBox="0 0 29.37 21.88"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon Only Button Snapshot should render with updated variant 1`] = `
<div>
  <button
    aria-label="Check Icon"
    class="css-qmynnk e183qit0"
    color="primary"
    shape="round"
  >
    <svg
      class="icon"
      fill="currentColor"
      height="18"
      viewBox="0 0 29.37 21.88"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon With Text Button Snapshot should render with large size 1`] = `
<div>
  <button
    class="css-1b9wjjc e183qit0"
    color="primary"
    shape="fill"
  >
    <span
      class="css-1p2jtab e1cefif80"
    >
      Test Text
       
    </span>
    <svg
      class="icon"
      fill="currentColor"
      height="12"
      viewBox="0 0 29.37 21.88"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon With Text Button Snapshot should render with small size 1`] = `
<div>
  <button
    class="css-1b9wjjc e183qit0"
    color="primary"
    shape="fill"
  >
    <span
      class="css-1p2jtab e1cefif80"
    >
      Test Text
       
    </span>
    <svg
      class="icon"
      fill="currentColor"
      height="12"
      viewBox="0 0 29.37 21.88"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon With Text Button Snapshot should render with updated color 1`] = `
<div>
  <button
    class="css-xh5nj e183qit0"
    color="secondary"
    shape="fill"
  >
    <span
      class="css-1p2jtab e1cefif80"
    >
      Test Text
       
    </span>
    <svg
      class="icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 29.37 21.88"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon With Text Button Snapshot should render with updated shape 1`] = `
<div>
  <button
    class="css-1vyo6ip e183qit0"
    color="primary"
    shape="fill"
  >
    <span
      class="css-1p2jtab e1cefif80"
    >
      Test Text
       
    </span>
    <svg
      class="icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 29.37 21.88"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon With Text Button Snapshot should render with updated text transform 1`] = `
<div>
  <button
    class="css-9tyo2z e183qit0"
    color="primary"
    shape="fill"
  >
    <span
      class="css-1p2jtab e1cefif80"
    >
      Test Text
       
    </span>
    <svg
      class="icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 29.37 21.88"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;

exports[`Button Icon With Text Button Snapshot should render with updated variant 1`] = `
<div>
  <button
    class="css-3a4sd6 e183qit0"
    color="primary"
    shape="fill"
  >
    <span
      class="css-1p2jtab e1cefif80"
    >
      Test Text
       
    </span>
    <svg
      class="icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 29.37 21.88"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z"
        transform="translate(-0.87 -3.06)"
      />
    </svg>
  </button>
</div>
`;
